tmail-backend/jmap/extensions/src/main/java/com/linagora/tmail/james/jmap/StandaloneEventAttendanceRepository.java

```diff
@@ -52,15 +67,79 @@ private Mono<AttendanceStatus> handleMissingEventAttendanceFlag(MessageId messag
     }
 
     @Override
-    public Publisher<Void> setAttendanceStatus(Username username, MessageId messageId,
-                                               AttendanceStatus attendanceStatus) {
+    public Publisher<CalendarEventReplyResults> setAttendanceStatus(Username username, AttendanceStatus attendanceStatus,
+                                                                    BlobIds eventBlobIds,
+                                                                    Optional<LanguageLocation> maybePreferredLanguage) {
         MailboxSession systemMailboxSession = sessionProvider.createSystemSession(username);
 
-        return Flux.from(messageIdManager.getMessagesReactive(List.of(messageId), FetchGroup.MINIMAL, systemMailboxSession))
-            .map(MessageResult::getMailboxId)
+        return getEnclosingMessageId(eventBlobIds).flatMap(messageId ->
```

Comment:
The assumption here that all Blobs will belong to the same message is wrong, as nothing in the JMAP API explicitly says that; thus, the PR needs a bit more work.